### PR TITLE
 chore: git sync cypress tests

### LINKED ISSUE #6012: Cypress tests
#### Tasks to-do before writing tests.
- Prepare DSLs that results in conflicts: while committing and merging
- Look into GitHub apis:
    - Create / delete repos
    - Create / delete deploy keys
- Update existing tests if they're breaking with new changes (for example debugger trigger)

________________________

### Tests

- [ ] Git connection: able to connect with the correct git user config: 
    - Generate SSH key works
        - Key based authentication: 
        - Upload / delete using GitHub api
    - Repo with / without existing files
    - Config: global vs local config: check the expected params:
        - Should we verify if the commit was made with correct creds
    - Public repos without auth (optional)
    - On connection success: verify the params (if local config is used, local config shouldn't be created if global config) in the /applications response
- [ ] Global config form
- [ ] Commit
    - Commit with or without push
        - Commit / push should be disabled if nothing to push or commit
        - Check params: message, doPush 
        - If done without push checked: we should see the push button
        - optional: if we need to verify the user detail on the commit and if the commit was finally pushed to remote using GitHub apis
    - Conflict state: try to create a conflict by importing via git repo in another app (also tests import flow via git)
        - Include verification for errors here during import
        - There should be an error for the conflict and a retry button

- [ ] Switch branch
    - Create a new branch:
        - Should have widgets created on parent branch
    - Switch between existing:
        - Verify if branch is getting switched by verifying the exclusive widgets on either branches
    - Show confirmation modal if we have any uncommitted changes

- [ ] Merge:
    - Check if two branches are mergeable: show status accordingly
    - Can't merge to itself
    - Switch branch: would also be tested in this flow

- [ ] Import :
    - Cases for valid and invalid repos
    - if we need cases to check repos across migrations

- [ ] Push
    - Check init state: disable if nothing to push
    - Check error state:
        - Retry after conflict 
- [ ] Pull
    - Check init state: disable if nothing to pull
    - Check error state:
        - Retry after conflict

### PR DESCRIPTION
**Pull request by rishabhsaxena**
## Description

Fixes #6012 

## Type of change

Add tests 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: add-git-connect-tests 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.74 **(-0.01)** | 37.06 **(-0.01)** | 35.8 **(0)** | 56.11 **(0)**
 :red_circle: | app/client/src/components/designSystems/appsmith/header/DeployLinkButton.tsx | 78.79 **(-1.86)** | 84.62 **(0)** | 16.67 **(0)** | 77.42 **(-1.89)**
 :red_circle: | app/client/src/pages/Editor/EditorAppName/NavigationMenuData.ts | 66.67 **(-0.54)** | 41.67 **(0)** | 6.25 **(0)** | 66.13 **(-0.54)**
 :red_circle: | app/client/src/pages/Editor/gitSync/QuickGitActions/index.tsx | 66.35 **(-1.3)** | 42.31 **(-1.69)** | 34.78 **(0)** | 68.37 **(-0.7)**
 :red_circle: | app/client/src/reducers/entityReducers/pageListReducer.tsx | 17.65 **(0)** | 25 **(0)** | 15 **(0)** | 19.57 **(-0.43)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**
 :sparkles: :new: | **app/client/src/utils/bootIntercom.ts** | **43.75** | **21.43** | **100** | **41.67**
 :sparkles: :new: | **app/client/src/utils/getQueryParamsObject.ts** | **80** | **100** | **100** | **77.78**
 :green_circle: | app/client/src/utils/helpers.tsx | 67.94 **(0.85)** | 34.88 **(0.31)** | 58.82 **(-1.56)** | 66.22 **(0.91)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

**Review by Parthvi12 — APPROVED** [review]

File: app/client/src/utils/bootIntercom.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { User } from "constants/userConstants";
```
Comment (IAmAnubhavSaini) [inline app/client/src/utils/bootIntercom.ts:1]: @haojin111 What is this file for?

**Review by IAmAnubhavSaini — APPROVED** [review]

**Pull request merged**